pkg/openstack/common.go

```diff
@@ -298,6 +298,13 @@ func EnsureEndpointConfig(
 						Labels:      util.MergeMaps(ed.Labels, map[string]string{serviceCertSelector: ""}),
 						Usages:      nil,
 					}
+					if svcName, ok := svc.Annotations["headless"]; ok {
```

Comment:
@stuggi this is an example of what I need for glance: if the annotation is not present everything will work as usual, but we can embed in a service the information about additional SNs that we want in the `CertificateRequest`.
I used `headless` to give you the idea of what I need to achieve, but I'm ok to have a different name or a different structure to get additional information from the `SVC` we're iterating here.

In addition, I'd like to minimize the amount of changes, and extending the function signature requires an update of the function call everywhere, so I think we can embed this information to the SVC, which is the source of truth to compute the certificate SubjectName(s).

Reply:
- should it be possible to use the annotation to be a list of hostnames? 
```
annoations:
  additionalSubjectNames:
    - first
    - second
    - third
```
- is it ok to expect those should always be added with:
```
string(fmt.Sprintf("*.%s.%s.svc", svcName, instance.Namespace)),
string(fmt.Sprintf("*.%s.%s.svc.%s", svcName, instance.Namespace,ClusterInternalDomain)),
```
or might there be the need to register any dns name. if so, should the `additionalSubjectNames` provide any names to register and we do not generate the fqdn here?

Reply:
Yes, I created a patch for lib-common where I can process a comma separated list of `additionalSubjectNames` set in the svc as an annotation from the service operator [1].
The idea is to remove:

```
string(fmt.Sprintf("*.%s.%s.svc", svcName, instance.Namespace)),
string(fmt.Sprintf("*.%s.%s.svc.%s", svcName, instance.Namespace,ClusterInternalDomain)),
```
because this detail can live in the annotation set by the service operator. Here we should just get the `[]string{}` of `additionalSubjectNames` and append them to the `CertificateRequest` field.

[1] https://github.com/openstack-k8s-operators/lib-common/pull/516

Reply:
I'll follow up in a bit here.